packages/data/src/components/with-dispatch/index.js

```diff
@@ -54,8 +54,11 @@ const withDispatch = ( mapDispatchToProps ) => createHigherOrderComponent(
 					// called, it is done only to determine the keys for which
 					// proxy functions should be created. The actual registry
 					// dispatch does not occur until the function is called.
-					const propsToDispatchers = mapDispatchToProps( this.props.registry.dispatch, props.ownProps );
+					const propsToDispatchers = mapDispatchToProps( this.props.registry.dispatch, props.ownProps, this.props.registry.select );
 					this.proxyProps = mapValues( propsToDispatchers, ( dispatcher, propName ) => {
+						if ( typeof dispatcher !== 'function' ) {
+							throw new TypeError( `${ propName } returned from mapDispatchToProps in withDispatch HOC must be a function` );
```

Comment:
I've tried to promote avoiding the acronym "HOC". I don't think it makes much sense to the majority, and in this specific instance I don't think it adds any value; the sentence still makes sense without it.

Reply:
I will update this message. I agree with your reasoning 👍